File: Persistence/Models/Project.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Persistence.Models
{
    public class ProjectModel
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public string Code { get; set; }
        public string Description { get; set; }
        public short ClientId { get; set; }
        public virtual ClientModel Client { get; set; }
        public long LeaderId { get; set; }
        public virtual UserModel Leader { get; set; }
        public long TypeId { get; set; }
        public virtual CatalogModel Type { get; set; }
        public long StatusId { get; set; }
        public virtual CatalogModel Status { get; set; }
        public long MethodologyId { get; set; }
        public virtual CatalogModel Methodology { get; set; }
        public bool PSP { get; set; }
        public bool Active { get; set; }
        public long UpdatedById { get; set; }
        public virtual UserModel UpdatedBy { get; set; }
        public DateTime UpdatedAt { get; set; }
        public virtual ICollection<ProjectTeamModel> Team { get; set; }
        public virtual ICollection<ActivityModel> Activities { get; set; }
    }
}

File: yitpro/src/app/models/project.ts
export class ProjectStatus {
  static canceled = "Cancelado";
}

export interface IProject {
  id: number;
  client: string;
  name: string;
  code: string;
  leader: string;
  leaderPhoto: string;
  type: string;
  methodology: string;
  status: string;
  active: boolean;
}

export interface IProjectTeam {
  id: number;
  user: string;
  userPhoto: string;
}

export interface IProjectFilter {
  name: string;
  code: string;
  clientId?: number | "";
  active?: boolean | "";
}

export interface IProjectDetail {
  id: number;
  clientId: number | "";
  name: string;
  code: string;
  leaderId: number | "";
  typeId: number | "";
  statusId: number | "";
  methodologyId: number | "";
  description: string;
  active: boolean;
  psp: boolean;
}

export class ProjectFilterValues implements IProjectFilter {
  name: string = "";
  code: string = "";
  clientId?: number | "" = "";
  active?: boolean | "" = "";

  constructor(init?: IProjectFilter) {
    Object.assign(this, init);
  }
}

export class ProjectFormValues implements IProjectDetail {
  id: number = 0;
  clientId: number | "" = "";
  name: string = "";
  code: string = "";
  leaderId: number | "" = "";
  typeId: number | "" = "";
  statusId: number | "" = "";
  methodologyId: number | "" = "";
  description: string = "";
  active: boolean = true;
  psp: boolean = false;

  constructor(init?: IProjectDetail) {
    Object.assign(this, init);
  }
}
